### PR TITLE
`fn rav1d_create_lf_mask_inter`: Hoist assert out of loop

### PR DESCRIPTION
**Pull request by rinon**

### CODE REVIEW

**Review by kkysen — CHANGES_REQUESTED** [review]
Whoops, I thought I sent this review last week!

File: src/lf_mask.rs [hunk 1]
```diff
@@ -522,10 +522,10 @@ pub(crate) fn rav1d_create_lf_mask_inter(
 
     if bw4 != 0 && bh4 != 0 {
         let mut level_cache_off = by * b4_stride + bx;
+        assert!(level_cache_off + (bh4 - 1) * b4_stride + bw4 < level_cache.len());
         for _y in 0..bh4 {
             for x in 0..bw4 {
                 let idx = level_cache_off + x;
-                assert!(idx < level_cache.len());
```
[the remark this answers is not in the document]
Comment (kkysen) [reply src/lf_mask.rs:528]: Actually, can you wait until #1273 merges, as that changes this (the inner `assert!` is no longer there).

[the remark this answers is not in the document]
Comment (rinon) [reply src/lf_mask.rs:528]: The assert wasn't getting removed, which was the issue and why I hoisted it. I'll re-check this after #1273 then.

**Comment by rinon** [comment]
I did, I'll re-open later if there's anything to gain here.

**Pull request closed without merging**